Unity-Technologies/ScriptableRenderPipeline
# Enable first stereo LWRP Graphics Tests using MockHMD

### Purpose of this PR
By adding stereo tests to the suite of LWRP Graphics Tests I hope to minimize situations where code changes in LWRP introduce problems for XR.  

---
### Release Notes
Adding split-screen stereo (MockHMD) tests to the LWRP Graphics Tests.  

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?
I compared commit c93bf43 of my dev branch against its parent master commit 044c2ab on Katana using recent trunk changeset eec7c1c162e9.  Links and observations follow.  Results for my dev branch match master except that digging into the logs show the 10 inconclusive tests on Mac, as intended per discussion above.  

* Master: 
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders/proj57-All%20Tests/builds/2914?ScriptableRenderLoop_branch=master&unity_branch=trunk&automation-tools_branch=master

* lwrp_mockhmd_dennisd: 
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders/proj57-All%20Tests/builds/2909?automation-tools_branch=master&ScriptableRenderLoop_branch=lwrp_mockhmd_dennisd&unity_branch=trunk


* Test Editor - Windows Build 
For both branches tests all passed but stdio contains error: 
[04:14:56] Aborting batchmode due to failure: [04:14:56] Fatal Error! guid is uninitialized

* Test PlayMode - Mac (Intel) 
For both branches 017_Lighting_Scene_DirectionalBaked failed in LWGraphicsTest. 
Additionally for lwrp_mockhmd_dennisd the 10 new stereo tests are inconclusive on Mac by design.  

* Test PlayMode - Windows
For both branches 017_Lighting_Scene_DirectionalBaked failed in LWGraphicsTest. 
Additionally for lwrp_mockhmd_dennisd the new 063_xr_Lighting_Scene_DirectionalBaked test failed,which is derived from 017_Lighting_Scene_DirectionalBaked.


**Manual Tests**: What did you do?
- [ YES ] Opened test project + Run graphic tests locally
- [ N/A ] Built a player
- [ N/A ] Checked new UI names with UX convention
- [ N/A ] Tested UI multi-edition + Undo/Redo
- [ N/A ] C# and shader warnings (supress shader cache to see them)
- Other: 
I ran LWRP Graphics Tests (including my new stereo tests) in the Editor on Windows and Mac and saw the expected behaviour, except for point light and spot light tests which I have toggled off; I'll file a bug about these.  

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
This PR creates automated tests which produced the new reference images that are included in this PR.  

Any test projects to go with this to help reviewers?
This PR adds scenes to the LWRP Graphics Tests project. 

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**: Low

---
### Comments to reviewers
This is my first SRP PR.  Feel free to add additional reviewers if applicable.  Also, all of my changes are under TestProjects, not in the packages... does this mean I do not need to update a CHANGELOG.md file?  

These new stereo tests are modified versions of existing non-stereo LWRP tests, so SRP staff should be familiar with their intended behaviour.  Since MockHMD changes FOV and split-screen images (left-eye-right-eye) introduce a different aspect ratio, I had to move some of the test content and/or camera within the scenes to ensure proper illustration of the techniques under test.  

Currently the stereo LWRP image capture mechanism results in blank output images on Mac, so I use Assume.That() to flag stereo Mac tests as inconclusive.  There is little emphasis on VR Mac support so this is not a pressing concern.  I verified that the LWRP Graphics Tests are considered green by Katana despite the presence of inconclusive stereo tests.  

Any additional stereo tests should have "_xr_" in the name, following the convention yyy_xr_z where yyy is the test number and z is descriptive of the test intent.
